vispy/visuals/colorbar.py

```diff
@@ -280,28 +284,27 @@ class ColorBarVisual(CompoundVisual):
 
     def __init__(self, cmap, orientation, size,
                  pos=[0, 0],
-                 label_str="",
+                 label="",
                  label_color='black',
                  clim=(0.0, 1.0),
                  border_width=1.0,
-                 border_color="black",
-                 **kwargs):
+                 border_color="black"):
 
-        self._label_str = label_str
-        self._label_color = label_color
         self._cmap = get_colormap(cmap)
         self._clim = clim
         self._pos = pos
         self._size = size
         self._orientation = orientation
 
-        self._label = TextVisual(self._label_str, color=self._label_color)
+        if not isinstance(label, TextVisual):
+            label = TextVisual(label, color=label_color)
+        self._label = label
```

Comment:
Do we need to override a given TextVisual's color with the given `label_color`? So if the user gives a `TextVisual` which is in `green` and also gives `label_color="red"`. Or should we just make clear in the docstring, that `label_color` is only used when `label` is a string?

Reply:
> Or should we just make clear in the docstring, that label_color is only used when label is a string?

That's what happens now, right? :)

Reply:
or do you mean the ticks below? Maybe these should use `label.color` instead?

Reply:
If so, we should also update the tick colors when the label setter is given a label object.

Reply:
So many options... We might need to add setters for all that stuff...

Reply:
I added a note in the `label` docstring that color is not used if it is a TextVisual. I think every Visual in VisPy needs to make their properties more available. Not sure this PR is going to be the place though. Might be something good for a hacktoberfest issue.